File: sections/54_geometry/problems/a_orientation/solution.py
import sys
def main():
    data=list(map(int,sys.stdin.buffer.read().split()))
    if not data: return
    t=data[0]; idx=1; out=[]
    for _ in range(t):
        ax,ay,bx,by,cx,cy=data[idx:idx+6]; idx+=6; v=(bx-ax)*(cy-ay)-(by-ay)*(cx-ax); out.append("LEFT" if v>0 else "RIGHT" if v<0 else "TOUCH")
    print("\n".join(out))
if __name__=="__main__": main()

File: sections/54_geometry/problems/a_orientation/solution.cpp
#include <bits/stdc++.h>
using namespace std; int main(){ios::sync_with_stdio(false);cin.tie(nullptr);int T;if(!(cin>>T))return 0;while(T--){long long ax,ay,bx,by,cx,cy;cin>>ax>>ay>>bx>>by>>cx>>cy;long long v=(bx-ax)*(cy-ay)-(by-ay)*(cx-ax);cout<<(v>0?"LEFT":v<0?"RIGHT":"TOUCH")<<'\n';}}
